tools/releaser/main.go

```diff
@@ -0,0 +1,304 @@
+			return err
+		}
+	}
+
+	workspace, err := os.OpenFile(workspacePath, os.O_RDWR, 0644)
+	if err != nil {
+		return err
+	}
+	defer workspace.Close()
+
+	if verbose {
+		fmt.Println("Preparing temporary WORKSPACE without gazelle directives.")
+	}
+	workspaceWithoutDirectives, err := getWorkspaceWithoutDirectives(workspace)
+	if err != nil {
+		return err
+	}
+
+	// reuse the open workspace file, so first we empty it and rewind
+	err = workspace.Truncate(0)
+	if err != nil {
+		return err
+	}
+	_ /* new offset */, err = workspace.Seek(0, os.SEEK_SET)
+	if err != nil {
+		return err
+	}
+
+	// write the directive-less workspace and update repos
+	if _, err := workspace.Write(workspaceWithoutDirectives); err != nil {
+		return err
+	}
+
+	if verbose {
+		fmt.Println("Running update-repos outputting to temporary file.")
+	}
+	cmd := exec.CommandContext(ctx, "bazel", "run", "//:gazelle", "--", "update-repos", "-from_file=go.mod", fmt.Sprintf("-to_macro=%s%%gazelle_dependencies", _tmpBzl))
+	cmd.Dir = os.Getenv("BUILD_WORKSPACE_DIRECTORY")
+	if out, err := cmd.CombinedOutput(); err != nil {
+		fmt.Println(string(out))
+		return err
+	}
+	defer os.Remove(tmpBzlPath)
+
+	// parse the resulting tmp.bzl for deps.bzl and WORKSPACE updates
+	if verbose {
+		fmt.Println("Parsing temporary bzl file to prepare deps.bzl and WORKSPACE modifications.")
+	}
+	maybeRules, workspaceDirectives, err := readFromTmp(tmpBzlPath)
+	if err != nil {
+		return err
+	}
+
+	// update deps
+	if verbose {
+		fmt.Println("Writing new deps.bzl")
+	}
+	if err := updateDepsBzlWithRules(depsPath, maybeRules); err != nil {
+		return err
+	}
+
+	// append WORKSPACE with directives at the end.
+	// except we cannot append directly because the earlier bazel //:gazelle run modified WORKSPACE
+	// so we truncate and seek to the beginning again before writing all of what we want
+	if verbose {
+		fmt.Println("Append WORKSPACE with directives")
+	}
+	_ /* new offset */, err = workspace.Seek(0, os.SEEK_SET)
+	if err != nil {
+		return err
+	}
+
+	// write the directive-less workspace and update repos
+	if _, err := workspace.Write(workspaceWithoutDirectives); err != nil {
+		return err
+	}
+	if _, err := workspace.Write(workspaceDirectives); err != nil {
+		return err
+	}
+
+	// cleanup before final gazelle run
+	//
+	// note that we also have a defer for os.Remove so it gets cleaned up if there are earlier errors.
+	// This defer will throw an error from this point on, but we're swallowing it anyways.
+	if verbose {
+		fmt.Println("Cleaning up temporary files")
+	}
+	if err := os.Remove(tmpBzlPath); err != nil {
+		return err
+	}
+
+	if verbose {
+		fmt.Println("Running final gazelle run, and copying some language specific build files.")
+	}
+	finalizationCommands := []struct {
+		cmd  string
+		args []string
+		then func() error
+	}{
+		{cmd: "bazel", args: []string{"run", "//:gazelle"}},
+		{cmd: "bazel", args: []string{"build", "//language/proto:known_imports"}, then: copyClosure(
```

Comment:
Optional: you can run `bazel build //language/proto:known_imports //language/proto:known_proto_imports //language/proto:known_go_imports` in one command and copy the 3 files later in their own loop, to avoid starting `bazel build` process 3 times